website/docs/d/aws_secret.html.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: "vault"
+page_title: "Vault: vault_aws_secret data source"
+sidebar_current: "docs-vault-datasource-aws-secret"
+description: |-
+  Reads AWS credentials from an AWS secret backend in Vault
+---
+
+# vault\_aws\_secret
+
+Reads AWS credentials from an AWS secret backend in Vault.
+
+~> **Important** All data retrieved from Vault will be
+written in cleartext to state file generated by Terraform, will appear in
+the console output when Terraform runs, and may be included in plan files
+if secrets are interpolated into any resource attributes.
+Protect these artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+## Example Usage
+
+```hcl
+resource "vault_aws_secret_backend" "aws" {
+  access_key = "AKIA....."
+  secret_key = "SECRETKEYFROMAWS"
+}
+
+resource "vault_aws_secret_role" "role" {
+  backend = "${vault_aws_secret_backend.aws.path}"
+  name = "test"
+  policy = <<EOT
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": "iam:*",
+      "Resource": "*"
+    }
+  ]
+}
+EOT
+}
+
+data "vault_aws_secret" "creds" {
```

Comment:
Maybe worth mentioning in a comment here that in practice the remainder of this would probably be in a separate config altogether, since it'd be weird to configure vault with credentials and then immediately read back again in the same config. :grinning:

Reply:
I can see use cases (e.g., write one set of creds to vault, have it issue a more restricted set of creds, have Terraform use those to set up your provider) but I can also see it causing confusion, so I added a comment. :)